File: Java/src/Cracking The Coding Interview/One_Two.java
package Ch1_InterviewQuestions;

import java.util.Scanner;

/*
 * Implement a function void reverse(char* str) in C or C++ (meh Java rules) which reverses a null terminated
 * string
 * I'll do similar but just in JAVA :^>
 */

public class One_Two {

	public static void main(String[] args) 
	{
		Scanner in = new Scanner(System.in);
		String input = in.nextLine();
		System.out.println(input);
		System.out.println(reverse(input));
	}
	
	public static String reverse(String input)
	{
		char[] temp = new char[input.length()];
		for(int i= 0; i < input.length(); i++)
		{
			temp[i] = input.charAt(input.length() - i - 1);
		}
		String reversed = new String(temp);
		return reversed;
	}

}

File: Python/src/Cracking the Coding Interview Problems/One_Two.py
"""1.2 Implement a function void reverse(char* str) in C or C++ which reverses a null-terminated
string. (Just reverse the damn string :^>)"""


def main():
    i = raw_input("Enter a string to be reversed: ")
    print i
    j = reverse_string(i)
    print j
    return


def reverse_string(string):
    j = [' ' for i in range(string.__len__())]
    for i in range(0, string.__len__()):
        j[i] = string[string.__len__() - 1 - i]
    string = ""
    string = string.join(j)
    return string

main()
